edx_api/enrollments/__init__.py

```diff
@@ -36,6 +46,56 @@ def get_student_enrollments(self):
         resp.raise_for_status()
         return Enrollments(resp.json())
 
+    def get_enrollments(self, course_id=None, username=None, usernames=None):
+        """
+        List all course enrollments.
+
+        Args:
+            course_id (str, optional): If used enrollments will be filtered to the specified course id.
+            username (str, optional): Enrollments will be filtered to this specific username.
+            usernames (list, optional): List of usernames to filter enrollments.
+        """
+        params = {'page_size': 1}
+        if course_id is not None:
+            params['course_id'] = course_id
+        if username is not None:
+            params['username'] = username
+        if usernames is not None:
+            params['username'] = ','.join(params.get('username', []) + usernames)
+
+        resp = self._get_list_request(params)
+        return Enrollments(resp.json()['results'])
```

Comment:
Won't this only retrieve the first page of results? What if there are multiple pages?

Reply:
Yes, I'm finishing adding the rest of the pages. Which brings me to my next question.
I wanted to do this with a generator but it seems like we would have to retrieve all enrollments and then return the result in this method. Is that OK?